Modules/posixmodule.c

```diff
@@ -12291,6 +12290,12 @@ win32_putenv(PyObject *name, PyObject *value)
         PyMem_Free(env);
         return NULL;
     }
+    if (wcslen(env) != (size_t)size) {
```

Comment:
This check is redundant with PyUnicode_AsWideCharString() check for embedded null characters, no?

Reply:
Ignore my comment. Since `PyUnicode_AsWideCharString(unicode, &size)` is used, the call doesn't reject embedded null characters... This API is subtle!